src/main/java/hudson/plugins/logparser/LogParserPublisher.java

```diff
@@ -161,16 +149,17 @@ public void perform(Run<?, ?> build, FilePath workspace, Launcher launcher, Task
 
     @Override
     public BuildStepDescriptor<Publisher> getDescriptor() {
-        return DescriptorImpl.DESCRIPTOR;
+        return Jenkins.getInstance().getDescriptorByType(LogParserPublisher.DescriptorImpl.class);
     }
 
+    @Extension @Symbol("logParser")
     public static final class DescriptorImpl extends
             BuildStepDescriptor<Publisher> {
-        public static final DescriptorImpl DESCRIPTOR = new DescriptorImpl();
-        private volatile ParserRuleFile[] parsingRulesGlobal = new ParserRuleFile[0];
+
+        private List<ParserRuleFile> parsingRulesGlobal = new ArrayList<ParserRuleFile>();
```

Comment:
better use a `DescribableList`, so it can be exposed by get() without risk for external changes, and use `replaceBy` in setter.

Reply:
Next iteration